scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -168,20 +168,20 @@ class AsyncFlowDefRunner(mode: CascadingMode) extends Writer {
         case Stop => ()
         case RunFlowDef(conf, fd, promise) =>
           try {
-            if (fd.getSinks.isEmpty) {
-              // These is nothing to do:
-              promise.success((id, JobStats.empty))
-            } else {
-              val ctx = ExecutionContext.newContext(conf.setScaldingFlowCounterValue(id))(fd, mode)
-              ctx.buildFlow match {
-                case Success(flow) =>
+            val ctx = ExecutionContext.newContext(conf.setScaldingFlowCounterValue(id))(fd, mode)
```

Comment:
Question is do we want to do `planTypedWrites` here or we want what you did instead?
I prefer `planTypedWrites` because it's more explicit and what you need here. Otherwise it's unclear that building `Flow` out of `FlowDef` in execution context is triggering side effect on `FlowDef`.

Alternatively comment about side effect would be good.

Reply:
I think it's fine to do `planTypedWrites` inside `buildFlow` since `buildFlow` is responsible to make sure everything in place for make flow happens.

Reply:
I like the way you did it, but this way you can break existing customers of `ExecutionContext`, can't you?

Reply:
since 0.18.x is not backward compatible and `ExecutionContext` is the internal class not for use outside of `scalding-core`, I think we pretty safe to make this change.